asserts/crypto.go

```diff
@@ -34,3 +42,129 @@ func generatePrivateKey() (*packet.PrivateKey, error) {
 	}
 	return packet.NewRSAPrivateKey(time.Now(), priv), nil
 }
+
+var openpgpConfig = &packet.Config{
+	DefaultHash: crypto.SHA256,
+}
+
+func signContent(content []byte, privKey *packet.PrivateKey) ([]byte, error) {
+	sig := new(packet.Signature)
+	sig.PubKeyAlgo = privKey.PubKeyAlgo
+	sig.Hash = openpgpConfig.Hash()
+	sig.CreationTime = time.Now()
+	sig.IssuerKeyId = &privKey.KeyId
+
+	h := openpgpConfig.Hash().New()
+	h.Write(content)
+
+	err := sig.Sign(h, privKey, openpgpConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	buf := bytes.NewBufferString("openpgp ")
+	enc := base64.NewEncoder(base64.StdEncoding, buf)
+	err = sig.Serialize(enc)
+	if err != nil {
+		return nil, err
+	}
+	enc.Close()
+
+	return buf.Bytes(), nil
+}
+
+func splitFormatAndDecode(formatAndBase64 []byte) (string, []byte, error) {
+	parts := bytes.SplitN(formatAndBase64, []byte(" "), 2)
+	if len(parts) != 2 {
+		return "", nil, fmt.Errorf("expected format and base64 data separated by space")
+	}
+	buf := make([]byte, base64.StdEncoding.DecodedLen(len(parts[1])))
+	n, err := base64.StdEncoding.Decode(buf, parts[1])
+	if err != nil {
+		return "", nil, fmt.Errorf("could not decode base64 data: %v", err)
+	}
+	return string(parts[0]), buf[:n], nil
+}
+
+// PublicKeyPrim offers the operations of a public key crypto primitive.
+type PublicKeyPrim interface {
+	// Fingerprint returns the key fingerprint.
+	Fingerprint() string
+	// Verify verifies signature is valid for content using the key.
+	Verify(content []byte, sig SignaturePrim) error
+}
+
+// TODO: have parse public key logic live here
+
+// SignaturePrim carries a signature crypto primitive.
+type SignaturePrim interface {
+	// KeyID() returns a suffix of the signing key fingerprint
+	KeyID() string
+}
+
+type signatureImpl struct {
```

Comment:
@niemeyer we could just do type Signature struct for, now and move to the 2nd model later if needed, anyway Signature differently from PublicKey is not something assertions themselves can be